### PR TITLE
test: add empty string and unicode edge cases to ValueTest

### PR DESCRIPTION
**Pull request by Tusharika725**
## Description
Resolves an inline `TODO` in `ValueTest.java` to add more test cases. 

Added edge cases for the `valueByteAsString()` method to ensure string encoding/decoding handles boundaries without data loss. Specifically added:
* An empty string boundary.
* A special character/Unicode boundary (testing whitespace, newlines, and a 4-byte emoji).

All local tests pass.

### CODE REVIEW

**Comment by jkwatson** [comment]
It would be extra-cool if you made this into a fuzz-test that would test a wider variety of possible inputs. I think there are examples of this in the codebase somewhere. (search for "fuzz")

**Comment by Tusharika725** [comment]
@jkwatson Thanks for the great suggestion! I've implemented a property-based fuzz test using JQF alongside the original deterministic tests. Let me know if you need any further adjustments!

File: api/all/src/test/java/io/opentelemetry/api/logs/ValueTest.java [hunk 1]
```diff
@@ -211,5 +217,39 @@ void valueByteAsString() {
...
+    String emptyBase64 = Value.of(emptyStr.getBytes(StandardCharsets.UTF_8)).asString();
+    byte[] decodedEmpty = Base64.getDecoder().decode(emptyBase64);
+    assertThat(new String(decodedEmpty, StandardCharsets.UTF_8)).isEqualTo(emptyStr);
+
+    // Test string with special characters
+    String specialStr = "hello \n \t 🌍 world!";
+    String specialBase64 = Value.of(specialStr.getBytes(StandardCharsets.UTF_8)).asString();
+    byte[] decodedSpecial = Base64.getDecoder().decode(specialBase64);
+    assertThat(new String(decodedSpecial, StandardCharsets.UTF_8)).isEqualTo(specialStr);
+  }
```
Comment (jack-berg) [inline api/all/src/test/java/io/opentelemetry/api/logs/ValueTest.java:221]: When testing repeated inputs / outputs, I prefer `@ParameterizedTest` style, e.g.: https://github.com/open-telemetry/opentelemetry-java/blob/main/sdk/common/src/test/java/io/opentelemetry/sdk/common/internal/GlobUtilTest.java#L17-L57

More readable and less verbose, especially when adding more test cases

File: api/all/src/test/java/io/opentelemetry/api/logs/ValueTest.java
Comment (jack-berg) [inline api/all/src/test/java/io/opentelemetry/api/logs/ValueTest.java]: I think we can probably remove this TODO
